Stratis.Bitcoin/Wallet/Wallet.cs

```diff
@@ -488,9 +502,14 @@ public bool IsSpendable()
 
         public Money SpendableAmount(bool confirmedOnly)
         {
-            // if this transaction hasn't been spent
-            if (this.SpendingDetails == null)
-            {            
+            // this method only returns a UTXO that has no spending output.
+            // if a spending output exists (even if its not confirmed) this 
+            // will return as zero balance.
+            
+            if (this.IsSpendable())
+            {
+                // if the 'confirmedOnly' flag is set check 
+                // that the UTXO has a block height.
```

Comment:
no need for this detail here, the isConfirmed method hides this implementation detail.